lib/database.cpp

```diff
@@ -0,0 +1,240 @@
+// SPDX-FileCopyrightText: 2021 Tobias Fella <fella@posteo.de>
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#include "database.h"
+
+#include <QtSql/QSqlDatabase>
+#include <QtSql/QSqlQuery>
+#include <QtSql/QSqlError>
+#include <QtCore/QStandardPaths>
+#include <QtCore/QDebug>
+#include <QtCore/QDir>
+
+#include "crypto/e2ee.h"
+#include "crypto/qolmsession.h"
+#include "crypto/qolminboundsession.h"
+
+//TODO: delete room specific data when leaving room
+
+using namespace Quotient;
+Database::Database()
+{
+    QSqlDatabase::addDatabase(QStringLiteral("QSQLITE"));
+    QString databasePath = QStandardPaths::writableLocation(QStandardPaths::AppDataLocation);
+    QDir(databasePath).mkpath(databasePath);
+    QSqlDatabase::database().setDatabaseName(databasePath + QStringLiteral("/database.db3"));
```

Comment:
Similar to the connection name - can the file be named somewhat more specific? It resides in the application data location, to which libQuotient doesn't really have the monopoly. Even `quotient.db3` would be better.